src/app/lib/utilities/setCookie/index.jsx

```diff
@@ -13,10 +13,27 @@ export const getCookieDomain = domain => {
   return domain;
 };
 
-const setCookie = (name, value, expires = COOKIE_EXPIRY) =>
-  Cookie.set(name, value, {
+const setCookie = ({ name, value, expires = COOKIE_EXPIRY, sameSite }) => {
+  const isHttps = window.location.protocol === 'https:';
+
+  // Modern browsers default sameSite value to Lax
+  // Setting sameSite='None' allows cookies will be sent in all contexts, i.e sending cross-origin is allowed.
+  // Setting sameSite='Strict' allows cookies to only set on first-party context
+  const sameSiteSecure = {
+    Strict: {
+      sameSite: 'Strict',
+    },
+    None: isHttps && {
+      sameSite: 'None',
+      secure: true, // When setting sameSite='None', the attribute secure=true must also be passed into the cookie otherwise the cookie will not be set.
+    },
+  };
+
+  return Cookie.set(name, value, {
     expires,
     domain: getCookieDomain(document.domain),
+    ...sameSiteSecure[sameSite],
```

Comment:
Does this mean passing 'Lax' as the `sameSite` argument will be ignored?

Reply:
Yeah, my initial intention here was that if a dev passed in any value aside from "None"/"Strict" it would just return undefined and then not pass the sameSiteSecure attributes which would mean the browser would use its own default.

However, I'm going to refactor this logic so that I am always passing in "secure" whenever it is https and sameSite to be explicitly Lax as a default.